src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,14 +1,67 @@
 package wolox.training.controllers;
-import org.springframework.stereotype.Controller;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.errors.book.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
+
+    @Autowired
+    private BookRepository bookRepository;
+
     @GetMapping("/greeting")
     public String greeting(@RequestParam(name="name", required=false, defaultValue="World") String name, Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
+
+    @PostMapping("/create")
```

Comment:
It is considered a good practice in Api Rest not to create specific routes for each method, that is, not to have a path "api/book/create", "api/book/delete", etc.

It is enough to have the different http method to differentiate one url from another. With the same http method you are saying if it is a cration, a search, an update or a deletion, it is not necessary to write it in the url.

I recommend using the following URLs:

POST api/books

DELETE api/books

GET api/ books/{isbn} (this is @PathVariable)

PUT api/books

Reply:
Can i ask that send me a book for update this? because i need the isbn for update, and the details of the book for update.

Reply:
Yes, I was wrong to tell you to just leave the isbn in the update

Reply:
Okay, so i resolved this problem